[15.0][IMP] sale_planner_calendar: Backport improvements from 16.0

carlosdauden opened the pull request:
@Tecnativa TT51739

cc @sergio-teruel @pilarvargas-tecnativa

pilarvargas-tecnativa committed:
[FIX] sale_planner_calendar: Remove partner name from default location address

By passing `True` to the `_display_address` method,
the company/partner name is omitted from the address
format, ensuring that only address details are displayed.
This change was to calculate the GPS location exactly.

TT51008

sergio-teruel committed:
[IMP] sale_planner_calendar: Refactor to use calendar.event model instead of sale.planner.calendar.event

carlosdauden committed:
[IMP] sale_planner_calendar: Merge files calendar.py and sale_planner_calendar_event.py in calendar_event.py

carlosdauden committed:
[IMP] sale_planner_calendar: Add sale_planner_state tracking and improve code

carlosdauden committed:
[IMP] sale_planner_calendar: Use calendar_event_type and remove sale_planner_calendar_event_profile

carlosdauden committed:
[IMP] sale_planner_calendar: New option to only update partner user_id if new_start is empty
TT51461

sergio-teruel committed:
[FIX] sale_planner_acalendar: Invalid field when user search events by week day in reassign wizard
TT51485

carlosdauden committed:
[IMP] sale_planner_calendar: Allow set new tags in reassign. Add partner_user_id field in calendar.event
Code improvements and translations.

TT51632

carlosdauden committed:
[IMP] sale_planner_calendar: Backport filters Past and Future

carlosdauden committed:
[IMP] sale_planner_calendar: Add ON CONFLICT to manage calendar_event_type with same name that sale_planner_calendar_event_profile

pilarvargas-tecnativa commented:
There is an error in the recurrence test. With the previous routing event model, when events were reassigned, a new recurrence was created for the modified events. In v16 I remember I had to modify the wizard logic in the migration because of the changes in the new version of Odoo. It is possible that with the reactorisation of the model change the bug in this version will come to light.

carlosdauden committed:
[IMP] sale_planner_calendar: Add invoice_date_due and calendar_issue_type on kanban view.
Add sale_planner_done_on_sale_confirm option.
Add products menu options.

TT51739

carlosdauden committed:
[IMP] sale_planner_calendar: Add reorder options in kanban view

sergio-teruel committed:
[IMP] sale_planner_calendar: Translate

CarlosRoca13 committed:
[IMP] sale_planner_calendar: Update records position when time changes

pilarvargas-tecnativa committed:
[FIX] sale_planner_calendar: Reset context to update new recurrence when reassigning an event

carlosdauden committed:
[FIX] sale_planner_calendar: Error when user timezone is empty

carlosdauden committed:
[IMP] sale_planner_calendar: Use base_event_id to compute is_base_recurrent_event

carlosdauden commented:
@sergio-teruel can you review?

pedrobaeza commented:
/ocabot merge major
